core/eth/reader.go

```diff
@@ -168,16 +168,18 @@ func (t *Reader) updateContractBindings(blsOperatorStateRetrieverAddr, eigenDASe
 		return err
 	}
 
+	var contractSocketRegistry *socketreg.ContractSocketRegistry
 	socketRegistryAddr, err := contractIRegistryCoordinator.SocketRegistry(&bind.CallOpts{})
 	if err != nil {
-		t.logger.Error("Failed to fetch SocketRegistry address", "err", err)
-		return err
-	}
-
-	contractSocketRegistry, err := socketreg.NewContractSocketRegistry(socketRegistryAddr, t.ethClient)
-	if err != nil {
-		t.logger.Error("Failed to fetch SocketRegistry contract", "err", err)
-		return err
+		//TODO: Temporarily warn, but continue without the socket registry
```

Comment:
Can you use a flag to control these?

Reply:
I could! Although it looks like I would need to update NewReader to take in a flag, and that requires updating flags for a lot of components like the controller, retriever, dataapi, apiserver, disperser, ... Do you think it is still the best way to process?

Reply:
It looks the easier way is to hold off the socket registry change until the contract is live. Is there anything blocked by not merging the change?

Reply:
I agree that's the easier way. To my understanding this PR unblocks this week's release but the contract is going live at the end of the sprint. 
Alternatively, we revert the changes that adds the socket registry and go ahead with the release; but then we wanted to do decentralized retrieval this sprint, and devops is using semvar to check correctness of the socket registry (migration from thegraph indexer)

Reply:
I was talking about not merging previous PR which added in this part of code.
It sounds nothing is blocked by this PR; all what you mentioned I'd say the blocker is the the deployment of the contract.

Reply:
A more appropriate way seems still to be some CLI flags, maybe defined similar to the graph indexer with default to false, passed in as a config etc. It'd be a standard approach as we add other contracts in the future.

To unblock the release for now, what if you add a local flag (not a CLI flag), to control flow here and the below GetOperatorSocket? Then when you actually ready to enable it, there is only one place to change, instead of editing code around.

Reply:
yep, that sounds great. I added a local variable `socketRegistryEnabled` and used that for control flow here and `GetOperatorSocket`